islandora.module

```diff
@@ -1685,8 +1690,12 @@ function islandora_object_access($op, $object, $user = NULL) {
  */
 function islandora_islandora_object_access($op, $object, $user) {
   module_load_include('inc', 'islandora', 'includes/utilities');
-
-  return islandora_namespace_accessible($object->id) && user_access($op, $user);
+  if ($object->state != 'A') {
+    return islandora_namespace_accessible($object->id) && user_access($op, $user) && user_access(ISLANDORA_ACCESS_INACTIVE_AND_DELETED_OBJECTS, $user);
```

Comment:
As a large implementor of Islandora, I understand from our devs here that this change, if merged, is going to cost us a significant amount of time to due to the configuration management issues Jordan noted above. So that's our biggest point of contention here at DGI ;).

Having both ways in paralell and deprecating the existing functionality over time is a reasonable approach.